### PR TITLE
drivers: sensor: iis2dh: Update driver to use DEVICE_DT_GET 

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

File: drivers/sensor/iis2dh/iis2dh_spi.c [hunk 1]
```diff
@@ -29,10 +29,9 @@ static struct spi_config iis2dh_spi_conf = {
 	.cs        = NULL,
```
Comment (galak) [inline drivers/sensor/iis2dh/iis2dh_spi.c:29]: Remove iis2dh_spi_conf as its not used anymore... Also do we need to fix the `operations` to match what is define here in `IIS2DH_SPI`

**Review by galak — CHANGES_REQUESTED** [review]
Remove unused iis2dh_spi_conf

File: drivers/sensor/iis2dh/iis2dh_spi.c [hunk 2]
```diff
@@ -100,28 +98,15 @@ stmdev_ctx_t iis2dh_spi_ctx = {
...
-	data->ctx = &iis2dh_spi_ctx;
-	data->ctx->handle = (void *)dev;
-
-#if DT_INST_SPI_DEV_HAS_CS_GPIOS(0)
-	/* handle SPI CS thru GPIO if it is the case */
-	data->cs_ctrl.gpio_dev = device_get_binding(
-		DT_INST_SPI_DEV_CS_GPIOS_LABEL(0));
-	if (!data->cs_ctrl.gpio_dev) {
-		LOG_ERR("Unable to get GPIO SPI CS device");
+	if (!device_is_ready(config->spi.bus)) {
```
Comment (galak) [inline drivers/sensor/iis2dh/iis2dh_spi.c:103]: Change `device_is_ready` to `spi_is_ready`

Comment (benjaminbjornsson) [reply drivers/sensor/iis2dh/iis2dh_spi.c:103]: Thanks!

File: drivers/sensor/iis2dh/iis2dh_spi.c (hunk 1 above)
Comment (benjaminbjornsson) [reply drivers/sensor/iis2dh/iis2dh_spi.c]: Removed, thanks!

Don't think I understand what you mean by fixing the `operations`?

Comment (galak) [reply drivers/sensor/iis2dh/iis2dh_spi.c]: So in there the spi_config is set to:
`	.operation = (SPI_OP_MODE_MASTER | SPI_MODE_CPOL |
		      SPI_MODE_CPHA | SPI_WORD_SET(8)),
`

Seems like we should have the same settings passed to `SPI_DT_SPEC_INST_GET` in `IIS2DH_SPI`

Comment (benjaminbjornsson) [reply drivers/sensor/iis2dh/iis2dh_spi.c]: I see, thank you!

**Review by galak — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**